File: 02684.py
for i in range(int(input())):
    s = input()

    cnt = {}
    cnt["TTT"] = 0
    cnt["TTH"] = 0
    cnt["THT"] = 0
    cnt["THH"] = 0
    cnt["HTT"] = 0
    cnt["HTH"] = 0
    cnt["HHT"] = 0
    cnt["HHH"] = 0

    for j in range(len(s) - 2):
        tmp = s[j:j+3]
        cnt[tmp] += 1

    for key, item in cnt.items():
        print(item, end=' ')
    print()

File: 02684.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
	int P; cin >> P;

	for (int i = 0; i < P; i++) {
		string s; cin >> s;

		int cnt[8] = { 0, 0, 0, 0, 0, 0, 0, 0 };
		for (int j = 0; j < s.length() - 2; j++) {
			string tmp = s.substr(j, 3);
			if (tmp == "TTT") cnt[0]++;
			else if (tmp == "TTH") cnt[1]++;
			else if (tmp == "THT") cnt[2]++;
			else if (tmp == "THH") cnt[3]++;
			else if (tmp == "HTT") cnt[4]++;
			else if (tmp == "HTH") cnt[5]++;
			else if (tmp == "HHT") cnt[6]++;
			else cnt[7]++;
		}

		for (int j = 0; j < 8; j++)
			cout << cnt[j] << ' ';
		cout << endl;
	}
}
